pkg/render/goldmane/component.go

```diff
@@ -123,23 +131,48 @@ func (c *Component) serviceAccount() *corev1.ServiceAccount {
 	}
 }
 
+// configMap returns a ConfigMap containing configuration for Goldmane that does not require a restart of the pod.
+// This is mounted as a file within the Pod, which can be dynamically reloaded when changed.
+func (c *Component) configMap() *corev1.ConfigMap {
```

Comment:
Nit, might be nice to name this dynamicConfigConfigMap or something like that (equally fine to leave it as well).